pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -191,7 +193,21 @@ func newEndpoint(address string, port uint32) (*endpoint.LbEndpoint, error) {
 // the endpoints for the cluster.
 func updateCluster(clusterName string, edsCluster *EdsCluster) {
 	// TODO: should we lock this as well ? Once we move to event-based it may not matter.
-	hostname, ports, labels := model.ParseServiceKey(clusterName)
+	var hostname string
+	var ports model.PortList
+	var labels model.LabelsCollection
+
+	// This is a gross hack but Costin will insist on supporting everything from ancient Greece
+	if strings.Index(clusterName, "outbound") == 0 { //new style cluster names
```

Comment:
if we switch to a format where only "." are used, like for instance in.proto.subset-dns-format, then you can detect the old version by looking at the "|".